### DISCUSSION TITLE
Environment variables configuration

### DISCUSSION THREAD

**Discussion by osvald0** [vercel/turborepo#8497] [Help]
### Summary

I have a monorepo project with:

Apps: mobile and web.
Packages: hooks, apis, models, etc.

I hooks I have some functions to fetch data. Hooks use some methods from apis.
In apis I'm trying to use some environment variables, but they are not loaded.
I tried with dotenv, but it threw an error.

Attached are images of my config.
What do you think the right way is to handle this case?

<img width="851" alt="Screenshot 2024-06-15 at 3 05 14 AM" src="https://github.com/vercel/turbo/assets/20977158/5becde31-8e5a-44b9-95d7-38350abef019">

![Screenshot 2024-06-15 at 1 48 44 AM](https://github.com/vercel/turbo/assets/20977158/4d4c2aa6-c207-4b72-b76b-577edd7c6438)

**Comment by DeveloperMindset123** [comment]
Hello, this looks like an import error on the build, Next.Js generally comes with .env file processor so separately installing dotenv package isn't necessary. A quick and dirty way to check if your environmental variables are loading as intended is to just console.log it and check what values are being outputted, that gives a solid indication of whether or not .env files are loading or not. If you want to know further, the following link might be of help. (https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables)

As for the second error, it seems there was another Github PR that helped clear up the confusion, the following is the link : https://github.com/vercel/next.js/discussions/54176. The culprit is most likely the use of "use strict" in your index.js file within the dist directory, as hooks are treated as client side components rather than server side and the fs module is a server side module. Additional link that may be helpful --> https://github.com/vercel/next.js/issues/52566#. Lastly, although it's not a big deal and more so depends on your personal preference, utilizing typescript over javascript throughout the entirety of the project is the general sentiment as it tends to be less error prone, but this is entirely up to you. (You can take a look at this link if you would like to gain a better understanding of why typescript is preferred over javascript, https://www.reddit.com/r/typescript/comments/sqswhb/is_it_better_to_just_use_ts_instead_of_js/)

I hope this helps, if not, let me know.

**Comment by osvald0** [comment]
I already checked them with a console.log before sending the question. That is the reason why I told the variables aren't loaded.

**Reply by DeveloperMindset123** [reply]
I see, I had this issue in the past, in some projects/instances my .env variables wouldn't load, I came up with a bootleg alternative to it, although it works, I think there might be better solutions out there. For starters, install the dotenv package using whichever package manager your using, and then use the following line:

```require('dotenv').config({ path: '/custom/path/to/.env' })```

Just replace the path within the string with whatever path your .env file is pointing to. Since chances are that for some reason your project may not automatically be picking up the path to the environment file. If you want to read up on it further, please refer to the following link : https://www.npmjs.com/package/dotenv. This should solve your issue with the dotenv side of the problem.

**Reply by osvald0** [reply]
I created a new fresh project using the [kitchen-sink](https://github.com/vercel/turbo/tree/main/examples/kitchen-sink) example. I can use env variables from the apps but not from the packages. The documentation is not clear about this tbh.

**Comment by askurat** [comment]
Also have the same issue. Any update on this? Or just how to load env vars from packages?

**Reply by osvald0** [reply]
Could you try adding the variable at the application level where you use the package?
